### ISSUE TITLE
Unhandled Exception: System.NullReferenceException: Object reference not set to an instance of an object.

### ISSUE THREAD

**Issue by lextudio-support** [obfuscar/obfuscar#533]
Getting exception right out of the gate. I just attempted to use SmartAssembly from RedGate and it was choking on compiler generated code from the use of _async_ and _await_ keywords. I'm guessing this tool has the same issue?

```
Loading project...Processing assembly: MyAssembly.Owin, Version=1.3.0.0, Culture=neutral, PublicKeyToken=null
Loading assemblies...Extra framework folders: Trying to resolve dependency: mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5
c561934e089
Trying to resolve dependency: mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089
Done.
Renaming:  fields...parameters...properties...events...methods...types...
Unhandled Exception: System.NullReferenceException: Object reference not set to an instance of an object.
   at Obfuscar.TypeKey.Matches(TypeReference type)
   at Obfuscar.Obfuscator.RenameType(AssemblyInfo info, TypeDefinition type, TypeKey oldTypeKey, TypeKey newTypeKey, TypeKey unrenamedTypeKe
y)
   at Obfuscar.Obfuscator.RenameTypes()
   at Obfuscar.Obfuscator.RunRules()
   at Obfuscar.Program.Main(String[] args)
```

```
<?xml version='1.0'?>
<Obfuscator>
  <Var name="InPath" value=".\input" />
  <Var name="OutPath" value=".\Obfuscator_Output" />

  <Module file="$(InPath)\MyAssembly.Owin.dll" />
</Obfuscator>
```

```
[CompilerGenerated]
private struct <ApplyResponseGrantAsync>d__46 : IAsyncStateMachine
{
    public int <>1__state;
    public AynonymousIdentityHandler <>4__this;
    public AsyncTaskMethodBuilder <>t__builder;
    private object <>t__stack;
    private TaskAwaiter <>u__$awaiter4b;
    public CookieOptions <cookieOptions>5__47;
    public IdentityCreatedContext <createdContext>5__4a;
    public bool <removeAnonymousCookie>5__49;
    public AuthenticationResponseRevoke <revoke>5__48;

    private void MoveNext();
    [DebuggerHidden]
    private void SetStateMachine(IAsyncStateMachine param0);
}
```

**Original Reported Date**: 2015-02-12T02:22:20.45-08:00
**Planned For Release**: 
**Original CodePlex ID**: 25

**Comment by lextudio-support** [comment]
**Copied from CodePlex without authors**:

It has little to do with async-await, as my test on other assemblies are OK.

If possible, you can send me the assembly via support@lextm.com. There should be something else. Or alternatively, you can download Obfuscar source code and debug further to see which line in `Obfuscar.TypeKey.Matches` method causes the exception.

**Original Posted Date**: 2015-02-14T22:18:29.373-08:00

**Comment by lextudio-support** [comment]
More info is required.

**Original Closed Date**: 2015-08-04T05:40:51.367-07:00
